tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBroker.java

```diff
@@ -40,9 +44,33 @@ public void onSuccessResponse(Response<TradeItAuthenticateResponse> response) {
                 TradeItSDK.getLinkedBrokerCache().cache(linkedBroker);
                 callback.onSuccess(linkedBrokerAccounts);
             }
+
+            @Override
+            public void onErrorResponse(TradeItErrorResult errorResult) {
+                linkedBroker.error = errorResult;
+                callback.onError(errorResult);
+            }
         });
     }
 
+    public void authenticateIfNeeded(final TradeItCallbackWithSecurityQuestion<List<TradeItLinkedBrokerAccount>> callback) {
+        if (this.error != null && this.error.requiresAuthentication()) {
+            this.authenticate(callback);
+        } else if (this.error != null && this.error.requiresRelink()) {
+            callback.onError(this.error);
+        } else {
+            callback.onSuccess(this.accounts);
+        }
+    }
+
+    private void setUnauthenticated() {
+        this.setError(new TradeItErrorResult(TradeItErrorCode.SESSION_EXPIRED, "Linked Broker initialized from keychain", Arrays.asList("This linked broker needs to authenticate.")));
```

Comment:
"Keychain" is iOS only. Android is called "Keystore". Also, users may not realize we are using the keystore to cache. Should change message to:
```java
TradeItErrorResult(TradeItErrorCode.SESSION_EXPIRED, "Authentication required", Arrays.asList("Linked broker was not authenticated after initializing.")
```